src/main/twirl/views/renderers/dateField.scala.html

```diff
@@ -5,27 +5,28 @@
 }
 
 @errorClass = @{if(errors.nonEmpty) "error" else ""}
+@invalidTrueFalse = @{if(errors.nonEmpty) "true" else "false"}
 
 <div class='form-group @errorClass'>
     @questions.get(field.name).map { q => <p class="question">@q.text</p> }
     @errors.map { e =>
-        <span class="error-message">@e.err</span>
+        <span class="error-message" role="alert">@e.err</span>
     }
 
     <div class='date-selector'>
         <div>
             <label for="@(field.name).day">Day</label>
-            <input type="text" title="@(field.name).day" id="@(field.name).day" name="@(field.name).day" value='@answers.get(s"${field.name}.day")' class="form-control">
+            <input type="text" title="@(field.name).day" id="@(field.name).day" name="@(field.name).day" value='@answers.get(s"${field.name}.day")' class="form-control" aria-label="@questions.get(field.name).map(_.text) Day" aria-invalid="@invalidTrueFalse">
```

Comment:
For each of these aria-labels for the form, wouldn't it be easier on the ear if the label was more precise? To my ear the trailing "Day" or "Month" etc is left hanging, and it would be better spoken english to say something like  "On which day do you want to hold the event?", and then "On which month?" and "On which year?"

However, given this is  meant to be a generic partial I would suggest "When do you want to hold the event? On which day?" is clearer.  This would just require changing the suffix to from "Day" to "On which day?" (and change as appropriate for each other date part.